src/app/Agent/AgentProbeTemplates.tsx

```diff
@@ -0,0 +1,442 @@
+        setUploadFilename('');
+        setUploading(false);
+      }
+      return !v;
+    });
+  };
+
+  const handleFileChange = (value, filename) => {
+    setFileRejected(false);
+    setUploadFile(value);
+    setUploadFilename(filename);
+  };
+
+  const handleUploadSubmit = () => {
+    if (!uploadFile) {
+      window.console.error('Attempted to submit template upload without a file selected');
+      return;
+    }
+    setUploading(true);
+    addSubscription(
+      context.api
+        .addCustomProbeTemplate(uploadFile)
+        .pipe(first())
+        .subscribe((success) => {
+          setUploading(false);
+          if (success) {
+            setUploadFile(undefined);
+            setUploadFilename('');
+            setModalOpen(false);
+            refreshTemplates();
+          }
+        })
+    );
+  };
+
+  React.useEffect(() => {
+    addSubscription(
+      context.notificationChannel
+        .messages(NotificationCategory.ProbeTemplateUploaded)
+        .subscribe((v) => refreshTemplates())
+    );
+  }, [addSubscription, context, context.notificationChannel, setTemplates]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.notificationChannel.messages(NotificationCategory.TemplateDeleted).subscribe((v) => refreshTemplates())
+    );
+  }, [addSubscription, context, context.notificationChannel, setTemplates]);
+
+  const handleUploadCancel = () => {
+    setUploadFile(undefined);
+    setUploadFilename('');
+    setModalOpen(false);
+  };
+
+  const handleDeleteButton = React.useCallback(
+    (rowData) => {
+      if (context.settings.deletionDialogsEnabledFor(DeleteWarningType.DeleteEventTemplates)) {
+        setRowDeleteData(rowData);
+        setWarningModalOpen(true);
+      } else {
+        handleDelete(rowData);
+      }
+    },
+    [context, context.settings, setWarningModalOpen, setRowDeleteData, handleDelete]
+  );
+
+  const handleWarningModalAccept = React.useCallback(() => {
+    handleDelete(rowDeleteData);
+  }, [handleDelete, rowDeleteData]);
+
+  const handleWarningModalClose = React.useCallback(() => {
+    setWarningModalOpen(false);
+  }, [setWarningModalOpen]);
+
+  const handleFileRejected = () => {
+    setFileRejected(true);
+  };
+
+  const handleSort = (event, index, direction) => {
+    setSortBy({ index, direction });
+  };
+
+  const authRetry = React.useCallback(() => {
+    context.target.setAuthRetry();
+  }, [context.target, context.target.setAuthRetry]);
+
+  if (errorMessage != '') {
+    return (
+      <ErrorView
+        title={'Error retrieving probe templates'}
+        message={
+          'This is commonly caused by the agent not being loaded/active, check that the target was started with the agent (-javaagent:/path/to/agent.jar) ' +
+          errorMessage
+        }
+        retry={isAuthFail(errorMessage) ? authRetry : undefined}
+      />
+    );
+  } else if (isLoading) {
+    return <LoadingView />;
+  } else {
+    return (
+      <>
+        <Card>
```

Comment:
Looks like the card lower part is being covered. Same for live probe view. Maybe we could add some spacing between helper text and table.

![Screenshot from 2022-10-21 16-41-55](https://user-images.githubusercontent.com/68053619/197285282-52c0f5cc-b74a-46f7-9b9d-69eb88830b18.png)

Reply:
Look like the card needs some sort of shadow or styling so it needs more spacing below.